settings/resource_settings.go

```diff
@@ -0,0 +1,153 @@
+package settings
+
+import (
+	"context"
+	"fmt"
+	"regexp"
+
+	"github.com/databricks/databricks-sdk-go/apierr"
+	"github.com/databricks/databricks-sdk-go/service/settings"
+	"github.com/databricks/terraform-provider-databricks/common"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+)
+
+// NewSettingsAPI creates SettingsAPI instance from provider meta
+func NewSettingsAPI(ctx context.Context, m any) SettingsAPI {
+	client := m.(*common.DatabricksClient)
+	return SettingsAPI{client, ctx}
+}
+
+// SettingsAPI exposes the Settings API
+type SettingsAPI struct {
+	client  *common.DatabricksClient
+	context context.Context
+}
+
+func (a SettingsAPI) getCurrentEtag(etag string) (string, error) {
+	res, err := a.Read(etag)
+	if err != nil {
+		// If settings is not set-up, we will get an error with the etag in it.
+		// This will happen, for instance, during the first ever call to GET.
+		return a.getEtagFromError(err)
+	}
+	return res.Etag, nil
+}
+
+func (a SettingsAPI) getEtagFromError(err error) (string, error) {
```

Comment:
Just so you know, not every exception will contain the metadata (which contains the etag), only certain 404 cases and the conflict case (409). Let's make sure we're not always anticipating metadata to be present on the error.

Reply:
Right now I am returning an error to the user if that happens. What should I do? As far as I understand, we need the Etag to do the operation, so returning the error seems the only option for me.

Reply:
409 (RESOURCE_CONFLICT) will get raised from the API if there are concurrent writes (patch/delete) to the setting. In such a case, the request would need to be retried by using the fresh etag that'll be present in the 409 response. I think we can do it like access mgmt API did (which also uses etags): https://github.com/databricks/terraform-provider-databricks/pull/2371/files#diff-7eef2f7ec147f91e0b6597efd71eeece37d9dbf900e83f03d8f5f00d6011b7e7R116. See `handleConflictAndUpdate`. I suggest we do that in PATCH and DELETE, it's like an additional retry on conflict. @hectorcast-db

Reply:
I always get the latest tag before updating. So there should be no conflict.

Reply:
Terraform executes things in parallel, so there is always a chance that some other resource may do an update.

Reply:
Updated